pkg/bmc/access_test.go

```diff
@@ -10,163 +10,188 @@ func init() {
-	T, H, P, A, err := getTypeHostPort("my.favoritebmc.com")
-	if err != nil {
-		t.Fatalf("unexpected parse error: %v", err)
-	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
-	}
-	if P != "" { // default is set in DriverInfo() method
-		t.Fatalf("unexpected port: %q", P)
-	}
-	if H != "my.favoritebmc.com" {
-		t.Fatalf("unexpected hostname: %q", H)
-	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
-	}
-}
-
-func TestParseHostPort(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("192.168.122.1:6233")
-	if err != nil {
-		t.Fatalf("unexpected parse error: %v", err)
-	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
-	}
-	if H != "192.168.122.1" {
-		t.Fatalf("unexpected hostname: %q", H)
-	}
-	if P != "6233" {
-		t.Fatalf("unexpected port: %q", P)
-	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
-	}
-}
-
-func TestParseHostPortIPv6(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("[fe80::fc33:62ff:fe83:8a76]:6233")
-	if err != nil {
-		t.Fatalf("unexpected parse error: %v", err)
-	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
-	}
-	if H != "fe80::fc33:62ff:fe83:8a76" {
-		t.Fatalf("unexpected hostname: %q", H)
-	}
-	if P != "6233" {
-		t.Fatalf("unexpected port: %q", P)
-	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
-	}
-}
-
-func TestParseHostNoPortIPv6(t *testing.T) {
-	// They either have to give us a port or a URL scheme with IPv6.
-	_, _, _, _, err := getTypeHostPort("[fe80::fc33:62ff:fe83:8a76]")
-	if err == nil {
-		t.Fatal("expected parse error")
-	}
-}
-
-func TestParseIPMIURL(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("ipmi://192.168.122.1:6233")
-	if err != nil {
-		t.Fatalf("unexpected parse error: %v", err)
-	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
-	}
-	if H != "192.168.122.1" {
-		t.Fatalf("unexpected hostname: %q", H)
-	}
-	if P != "6233" {
-		t.Fatalf("unexpected port: %q", P)
-	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
-	}
-}
-
-func TestParseIPMIURLNoSep(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("ipmi:192.168.122.1")
-	if err != nil {
-		t.Fatalf("unexpected parse error: %v", err)
-	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
-	}
-	if H != "192.168.122.1" {
-		t.Fatalf("unexpected hostname: %q", H)
-	}
-	if P != "" {
-		t.Fatalf("unexpected port: %q", P)
-	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
-	}
-}
-
-func TestParseIPMIURLNoSepPort(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("ipmi:192.168.122.1:6233")
-	if err != nil {
-		t.Fatalf("unexpected parse error: %v", err)
-	}
-	if T != "ipmi" {
-		t.Fatalf("unexpected type: %q", T)
-	}
-	if H != "192.168.122.1" {
-		t.Fatalf("unexpected hostname: %q", H)
-	}
-	if P != "6233" {
-		t.Fatalf("unexpected port: %q", P)
-	}
-	if A != "" {
-		t.Fatalf("unexpected path: %q", A)
+func TestParse(t *testing.T) {
+	for _, tc := range []struct {
+		Scenario    string
+		URL         string
```

Comment:
Maybe name this something like `bmcAddr`, since we're about to change the output to an actual URL object and this could get confusing?

Reply:
Done